modules/augment/src/flip.cpp

```diff
@@ -3,35 +3,31 @@
 // of this distribution and at http://opencv.org/license.html.
 
 #include "precomp.hpp"
-namespace cv {
-namespace augment {
+namespace cv { namespace augment {
 
+FlipHorizontal::FlipHorizontal() {}
 
-    FlipHorizontal::FlipHorizontal() {}
-
-    void FlipHorizontal::image(InputArray _src, OutputArray _dst)
-    {
-        cv::flip(_src, _dst, 1);
-    }
-
-    Point2f FlipHorizontal::point(InputArray image,const Point2f& src)
-    {
-        Mat imageM = image.getMat();
-        return Point2f(imageM.cols - 1 - src.x, src.y);
-    }
+void FlipHorizontal::image(InputArray _src, OutputArray _dst)
+{
+    cv::flip(_src, _dst, 1);
+}
 
+Point2f FlipHorizontal::point(const Point2f& src)
+{
+    return Point2f(srcImageCols - 1 - src.x, src.y);
+}
 
-    FlipVertical::FlipVertical() {}
 
-    void FlipVertical::image(InputArray _src, OutputArray _dst)
-    {
-        cv::flip(_src, _dst, 0);
-    }
+FlipVertical::FlipVertical() {}
 
-    Point2f FlipVertical::point(InputArray image,const Point2f& src)
-    {
-        Mat imageM = image.getMat();
-        return Point2f(src.x, imageM.rows - 1 - src.y);
-    }
+void FlipVertical::image(InputArray _src, OutputArray _dst)
+{
+    cv::flip(_src, _dst, 0);
 }
+
+Point2f FlipVertical::point(const Point2f& src)
+{
+    return Point2f(src.x, srcImageRows - 1 - src.y);
 }
```

Comment:
I want to get rid of `srcImageRows` and `srcImageCols` from the fileds members. I think it's OK if we pass image as an InputArray: `Point2f point(InputArray image, const Point2f& src)`.

Reply:
by doing that we will have to generate the rotation matrix everytime because we need these information to generate it.. and if we take the image as input to init to generate it.. I think it's much more convenient to store the needed info about it rather than needing to extract them each time..
don't you think?

Reply:
`srcImageRows` and `srcImageCols` are not actually used for generation. All the entries of them might be replaced with `.rows` and `.cols` from `src` images.

Reply:
BTW, I just figured out that we must regenerate rotation matrix if input image size changed (because center changed). So we need to manage it as well.

Reply:
they are used to generate the rotation matrix to specify the center of rotation